components/cd-disclosure/cd-disclosure.js

```diff
@@ -0,0 +1,81 @@
+(function (Drupal) {
+  'use strict';
+
+  /**
+   * @see https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/
+   */
+  Drupal.behaviors.cdDisclosure = {
+    idPrefix: 'cd-disclosure--',
+
+    attach: function (context, settings) {
+      // Initialize toggable dropdown.
+      this.initializeDisclosures();
+    },
+
+    /**
+     * Set up all Disclosure widgets found in the DOM.
+     *
+     * For our purposes, the `disclosure` element is the wrapped content itself
+     * and the button to toggle it is generated automatically by this script.
+     */
+    initializeDisclosures: function () {
+      let disclosures = document.querySelectorAll('.cd-disclosure');
+
+      // All behavior is controlled by data attributes.
+      disclosures.forEach((disclosure) => {
+        // Set a random ID if none is found.
+        let disclosureId = disclosure.getAttribute('id') || this.idPrefix + Math.floor(Math.random(1) * 100000000);
+
+        // Apply ID if it wasn't there already.
+        if (!disclosure.getAttribute('id')) {
+          disclosure.setAttribute('id', disclosureId);
+        }
+
+        // Determine where to place the button.
+        let positionAfter = disclosure.dataset.cdDisclosureInsert === 'after' || false;
+
+        // Determine whether content is shown by default. We consider this the
+        // edge-case so omitting the attribute assumes it is hidden.
+        let contentVisible = disclosure.dataset.cdDisclosureShowByDefault === 'true' || false;
+
+        // Extract button label
+        let buttonText = disclosure.dataset.cdDisclosureLabel || 'Toggle content';
+
+        // Create the toggler button.
+        let disclosureButton = document.createElement('button');
+        disclosureButton.setAttribute('aria-controls', disclosureId);
+        disclosureButton.setAttribute('aria-expanded', contentVisible);
```

Comment:
I think the button could benefit from an `ID` derived from the `disclosureId` so it can easily be targeted by other javascript (thought that can be done with a `button[aria-controls="ID"]`). (Use cases: RW and UNO which have extra logic for the scrolling to the top of the content).

Reply:
I'd assumed the button selector you mentioned would be how these things get targeted, but we can add another ID when we create them no problem 👍🏼

Reply:
I'm using `cd-for--ID` where it's the full ID of the other element. I like this as a convention we could use in the future, where you can always "find" the toggler ID and vice versa. I took inspiration from HTML `for` attributes on form elements `<label for="my-html-id">Label</label>`